go-controller/pkg/ovn/controller/admin_network_policy/baseline_admin_network_policy.go

```diff
@@ -33,6 +46,545 @@ func (c *Controller) processNextBANPWorkItem(wg *sync.WaitGroup) bool {
 	return true
 }
 
+// NOTE: A cluster can have only BANP at a given time as defined by upstream KEP.
+const (
+	BANPFlowPriority         = 1750 // down to 1651 (both inclusive, note that these ACLs will be in tier3)
+	BANPMaxPriorityPerObject = 100
+	BANPIngressPrefix        = "BANPIngress"
+	BANPEgressPrefix         = "BANPEgress"
+	BANPExternalIDKey        = "BaselineAdminNetworkPolicy" // key set on port-groups to identify which BANP it belongs to
+)
+
+// TODO: Double check how empty selector means all labels match works
+type baselineAdminNetworkPolicyState struct {
```

Comment:
AFAIK BANP is just ANP with the pre-defined prio, no Skip action, and different tier.
do you think we can just reuse ANP code for it, if we use a single internal struct like `adminNetworkPolicyState` adding `tier` field, setting pre-defined prio for BANP and converting action to nbdb action?

Reply:
yeah it should be doable! let me give it a try; if we can cut all the cloneCode by half that's not a bad idea..

Reply:
OK so this is nice haha, I managed to remove a LOT of duplicate functions and code that should make reviewing BANP easier. I might be able to consolidate the main `syncs` better maybe in future let's see. First let me fix all the tests and cleanup a bit.

Reply:
so I have cleanedup most of BANP and made common functions that both ANP & BANP can reuse. It should be better than before..I still left some redundancy for being able to follow the logic easily without abstracting it too much. resolving this comment now.